crates/weaver_resolved_schema/src/signal.rs

```diff
@@ -53,8 +54,6 @@ pub struct Event {
     /// References to attributes defined in the catalog.
     #[serde(skip_serializing_if = "Vec::is_empty")]
     attributes: Vec<AttributeRef>,
-    /// The domain of the event.
-    domain: String,
```

Comment:
> By removing this field, I believe we can no longer load and parse previous versions of the registry.

This was never a "top" level value and no "log" based events had ever defined in yaml or supported in semconv (or build tools). And the previous event "domain" was defined as an "attribute" associated with the event only and not a specific. As far as I can tell this is also never handled for anything else today either -- I'll go and recheck build-tools, but I don't recall seeing that there.

> Another similar issue is the renaming of the event_name field to name. While using name consistently across the signals is preferable, we should implement a serde alias for the name field to allow event_name for backward compatibility.
Nice catch -- I'll add that back

> attributes was also an optional field in the span events. Since attributes are no longer part of this generic event, it introduces another source of incompatibility with previous versions of the registries.
??? Optional Attributes are part of the standard event and I believe are still present (as (apart from the `event_name` to `name`) I just renamed the existing `SpanEvent` to `Event`...

Reply:
> `domain` removal

Just checked build-tools and prior version of weaver, the `domain` is only specified on the `Event` (log based event) which is not supported in any version of semconv yet, so I think this is fine.

> `attributes` removal

I just checked the `Event` and `SpanEvent are identical except for the `event_name`, so (I think) this is fine also.

Reply:
It's not about whether it's a fine, it's about whether using weaver on previous versions of semconv crashes or continues working as it used to.

I think you need to keep those things and deprecate them / ensure they get removed from semconv before you can remove them from weaver.

Reply:
I don't  see any usages of domain in semconv and it was never supported by the build-tools, so I think it should be fine to remove right away

Reply:
Please disregard my previous comment on `attributes`; they are indeed included at the group level. At this point, I believe we can simply add an alias for the `name` field to support the previous `event_name`, and then add a test to verify this.

Reply:
> It's not about whether it's a fine, it's about whether using weaver on previous versions of semconv crashes or continues working as it used to.
> 
> I think you need to keep those things and deprecate them / ensure they get removed from semconv before you can remove them from weaver.

Thats what I was trying to describe, `domain` has never been defined / used in any previous versions of sem-conv (at least as far as I can tell) and it's not been defined in the old build tools. It's just that weaver defined a `Event` definition that has never been used and I'm now reusing (and making active) that stuct.

Reply:
I've added several of these as topics for wednesday's meeting, as I added the #serde rename for the `event_name` and it didn't seem to do what I expected (ie. all of the json tests continued to pass), and as we don't have any "markdown" or "codegen" tests for "events" I suspect that the existing generations that (might) reference `event_name` won't work with this change alone.

Reply:
@MSNev, this is because you should use the annotation `#[serde(alias = "event_name")]` instead of the `rename` annotation. The `rename` annotation instructs Serde to use only the name specified in the `rename` statement, instead of using both the field name and the name specified in the `rename` annotation. The `alias` annotation does exactly what you need.

Reply:
FYI - here's where event domain was removed: https://github.com/open-telemetry/semantic-conventions/pull/473